File: app/src/main/java/com/example/providermoduledemo/sample/SampleViewBindItemBuilder.java
package com.example.providermoduledemo.sample;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.providermoduledemo.viewmodel.ViewBindItemLIRTGroupStyle0;
import com.example.providermoduledemo.viewmodel.ViewBindItemLIRTGroupStyle1;
import com.qq.reader.bookstore.CommonViewHolder;
import com.qq.reader.zebra.BaseViewBindItem;
import com.qq.reader.zebra.inter.IViewBindItemBuilder;

import java.util.ArrayList;
import java.util.List;

/**
 * ViewBindItemBuilder 示例
 */
public class SampleViewBindItemBuilder implements IViewBindItemBuilder<SampleResultBean> {
    @Override
    public List<BaseViewBindItem<?, ? extends RecyclerView.ViewHolder>> buildViewBindItem(@NonNull SampleResultBean data) {
        List<BaseViewBindItem<?, ? extends RecyclerView.ViewHolder>> viewBindItemList = new ArrayList<>();
        List<SampleResultBean.Item> dataList = data.getList();
        if (dataList == null) return viewBindItemList;
        for (SampleResultBean.Item item : dataList) {
            BaseViewBindItem<SampleResultBean.Item, CommonViewHolder> bindViewItem = null;
            switch (item.getStyle()) {
                case 0:
                    bindViewItem = new ViewBindItemLIRTGroupStyle0();
                    break;
                case 1:
                    bindViewItem = new ViewBindItemLIRTGroupStyle1();
                    break;
            }
            if (bindViewItem != null) {
                bindViewItem.setData(item);
                viewBindItemList.add(bindViewItem);
            }
        }
        return viewBindItemList;
    }
}

File: app/src/main/java/com/example/providermoduledemo/sample/SampleViewBindItemBuilder.kt
package com.example.providermoduledemo.sample

import androidx.recyclerview.widget.RecyclerView
import com.chad.library.adapter.base.BaseViewHolder
import com.example.providermoduledemo.viewmodel.ViewBindItemLIRTGroupStyle0
import com.example.providermoduledemo.viewmodel.ViewBindItemLIRTGroupStyle1
import com.qq.reader.provider.BaseViewBindItem
import com.qq.reader.provider.inter.IViewBindItemBuilder


/**
 * ViewBindItemBuilder 构建示例
 */
class SampleViewBindItemBuilder : IViewBindItemBuilder<SampleResponseBean> {
    override fun buildViewBindItem(data: SampleResponseBean): List<BaseViewBindItem<*, out RecyclerView.ViewHolder>> {
        val viewBindItemList = mutableListOf<BaseViewBindItem<SampleResponseBean.Item, BaseViewHolder>>()
        if (data.list == null) {
            return viewBindItemList
        }
        for (item in data.list!!) {
            var bindViewItem : BaseViewBindItem<SampleResponseBean.Item, BaseViewHolder>? = null
            when (item.style) {
                0 -> bindViewItem = ViewBindItemLIRTGroupStyle0()
                1 -> bindViewItem = ViewBindItemLIRTGroupStyle1()
            }
            bindViewItem?.let {
                it.data = item
                viewBindItemList.add(it)
            }
        }
        return viewBindItemList
    }
}